### ISSUE TITLE
Linq in ComboBox

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#179]
<sup>Issue created by Luiz Felipe Beneton as [Bitbucket Issue #&#8203;143](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143) on 2017.05.11 16:01.</sup>
Hi there, would you mind removing Linq dependency on ComboBox and AutoCompleteComboBox?

Thanks!

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.05 10:11](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143#comment-37371504), @SimonDarksideJ commented:</sup>
Any particular reason?  Linq saves time by removing the need for a 0(n) sort as it's better at logical searching.

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.05 12:57](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143#comment-37375615), Luiz Felipe Beneton commented:</sup>
I can have made a mistake, but I think a non-Linq filter is faster and allocates less memory:


Using the normal Prune method from AutoCompleteComboBox in a list with 100 items, Unity profiler tells me it allocated 18.4kb and took 13.96ms to run.

Using the method below, it allocated 2.9kb and took 1.34ms to run. All allocations coming from "ToLower()" which could be reduced with there was a list of all strings in lower case already. 

I hope that helps.

Cheers!



```c#

private void PruneItems(string currText)
    {
        string _currText = currText.ToLower();

        for (int i = _panelItems.Count - 1; i >= 0 ; i--)
        {
            string _item = _panelItems[i].ToLower();
            if (!_item.Contains(_currText))
            {
                _panelItems.RemoveAt(i);
                _prunedPanelItems.Add(_item);
            }
        }
    }
```

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.05 16:00](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143#comment-37380175), @SimonDarksideJ commented:</sup>
I like Stats.

If you have evidence that it would be quicker this way, then I'm more than happy to implement it.  or possibly have it as another indexing option at the very least.  Considering the frequency this runs, having a lowercase index would certainly be a lot faster.

I will look in to it.

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.05 19:43](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143#comment-37384389), Luiz Felipe Beneton commented:</sup>
Well, in my tests I got the number above. I would say that proves is faster :)

But you should run your own tests to be sure about it.

Cheers!

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.10 14:33](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/143#comment-37501349), @SimonDarksideJ commented:</sup>
Resolves #143

Performance tweaked the existing Linq method and at the request of #143 , added a new array sort alternative.


Referenced by 9ce211c5fc62

**Comment by SimonDarksideJ** [comment]
<sup>On 2017.06.10 14:33 @SimonDarksideJ modified issue:</sup>
**status** changed `new` → `resolved`
